Noni_MyLinkedList

noni-mizu opened the pull request:
Incomplete, will finish tomorrow.

Commit:
Noni_MyLinkedList

Incomplete, will finish tomorrow.

rsalota commented on src/com/seattletechinterviews/datastructures/MyLinkedList.java:
```diff
@@ -1,27 +1,92 @@
+			return null;
+		}// if location is the first node
+		else if (head.value == location){
+			return head.value;
```
You have to remove the element - this is just returning the value.

rsalota commented on src/com/seattletechinterviews/datastructures/MyLinkedList.java:
```diff
@@ -1,27 +1,92 @@
 	 * @param location
 	 */
-	public int removeAt(int location);
+	public int removeAt(Node head, int location){
```
You cannot change the signature of the function - you should implement this interface.

rsalota commented on src/com/seattletechinterviews/datastructures/MyLinkedList.java:
```diff
@@ -1,27 +1,92 @@
+			return head.value;
+		}
+		// if location is in the middle
+		while (head.next != null){
```
Typically you should store the head in a temporary location - like current.

rsalota commented on src/com/seattletechinterviews/datastructures/MyLinkedList.java:
```diff
@@ -1,27 +1,92 @@
+			if (headMoved >= 2){
+				previous = previous.next;
+			}
+			if (head.value == location){
```
The ask is not to search for the value - the ask is to remove the node at the given location and return the value at the location.

rsalota commented on src/com/seattletechinterviews/datastructures/MyLinkedList.java:
```diff
@@ -1,27 +1,92 @@
+			tmp = tmp.next;	
+		}
+		// add new node to end of list
+		if (tmp.next == null){
```
you don't need this end condition - as you are in the loop and are exiting it when tmp.next is null

rsalota commented on src/com/seattletechinterviews/datastructures/MyLinkedList.java:
```diff
@@ -1,27 +1,92 @@
 	 * Count the total number of elements in the linkedlist.
 	 */
-	public int count();
+	public int count(Node head){
```
Don't change the signature

rsalota commented on src/com/seattletechinterviews/datastructures/MyLinkedList.java:
```diff
@@ -1,27 +1,92 @@
+	public int count(Node head){
+		// start at one, since start at head
+		int counter = 1;
+		while(head.next != null){
```
what if head is null?

rsalota commented on src/com/seattletechinterviews/datastructures/MyLinkedList.java:
```diff
@@ -29,19 +94,83 @@
-	public int insertNth(int value,int location);
-	
+	 // Return the node.value? I'm assuming that.
+	public int insertNth(int value, int location, Node head){
```
Dont change the signature

rsalota commented on src/com/seattletechinterviews/datastructures/MyLinkedList.java:
```diff
@@ -29,19 +94,83 @@
+		Node tmp = head;
+		// need to count locations as we go
+		// starting with 0 as head
+		int counter = 0;
```
How do you handle the case where - he wants to insert before the head ?

rsalota commented on src/com/seattletechinterviews/datastructures/MyLinkedList.java:
```diff
@@ -29,19 +94,83 @@
+		// starting with 0 as head
+		int counter = 0;
+		while(tmp.next != null){
+			if (counter == location){
```
You need to allocate memory :-)

rsalota commented on src/com/seattletechinterviews/datastructures/MyLinkedList.java:
```diff
@@ -29,19 +94,83 @@
 	 * Delete the entire linked list.
 	 */
-	public void deleteList();
+	public void deleteList(Node head){
```
Cannot change the function signature

Commit:
Reformatted / corrections

Implemented interface into my own class, corrected some methods with comments, put method signatures back to how they should be.

Commit:
Added removeDuplicates()

Commit:
Completed mergeSort(), created additional class methods
